thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -211,6 +213,72 @@ public Response computeSeverity(@NotNull @QueryParam("collection") String collec
+   * @param id new anomaly function id
+   * @param startTimeIso start time of replay
+   * @param endTimeIso end time of replay
+   * @param isForceBackfill whether force back fill or not
+   * @param speedup whether use speedUp or not
+   * @param userDefinedPattern tuning paramter, user defined pattern can be "UP", "DOWN", or "UP&DOWN"
+   * @param sensitivity sensitivity level for initial tuning
+   * @param fromAddr email notification from address
+   * @param toAddr email notification to address
+   * @param subject subject of email
+   * @param teHost thirdeye host
+   * @param smtpHost smtp host
+   * @param smtpPort smtp port
+   * @param phantomJsPath phantomJSpath
+   * @return
+   */
+  @POST
+  @Path("/{id}/notifyreplaytuning")
+  public Response triggerReplayTuningAndNotification(@PathParam("id") @NotNull final long id,
+      @QueryParam("start") @NotNull String startTimeIso,
+      @QueryParam("end") @NotNull String endTimeIso,
+      @QueryParam("force") @DefaultValue("false") String isForceBackfill,
+      @QueryParam("speedup") @DefaultValue("false") final Boolean speedup,
+      @QueryParam("userDefinedPattern") @DefaultValue("UP") String userDefinedPattern,
+      @QueryParam("sensitivity") @DefaultValue("MEDIAN") final String sensitivity,
+      @QueryParam("from") String fromAddr,
+      @QueryParam("to") String toAddr, @QueryParam("subject") String subject,
+      @QueryParam("teHost") String teHost, @QueryParam("smtpHost") String smtpHost,
+      @QueryParam("smtpPort") int smtpPort,
+      @QueryParam("phantomJsPath") String phantomJsPath) {
+
+    // run replay, update function with jobId
+    long jobId;
+    try {
+      Response response = generateAnomaliesInRange(id, startTimeIso, endTimeIso, isForceBackfill, speedup, false);
+      Map<Long, Long> entity = (Map<Long, Long>) response.getEntity();
+      jobId = entity.get(id);
+    } catch (Exception e) {
+      return Response.ok("Failed to start replay!").build();
+    }
+
+    JobStatus jobStatus = detectionJobScheduler.waitForJobDone(jobId);
+    if (!jobStatus.equals(JobStatus.COMPLETED)) {
+      return Response.ok("Replay job error with job status: {}" + jobStatus).build();
+    } else {
+      LOG.info("Replay completed");
+    }
+
+    // create initial tuning and apply filter
+    Response initialAutotuneResponse = initiateAlertFilterAutoTune(id, startTimeIso, endTimeIso,"AUTOTUNE", 10, "", "");
+    updateAlertFilterToFunctionSpecByAutoTuneId(Long.valueOf(initialAutotuneResponse.getEntity().toString()));
+    LOG.info("Initial alert filter applied");
+
+    // send out email
+    long startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso).getMillis();
+    long endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso).getMillis();
+    emailResource.generateAndSendAlertForFunctions(startTime, endTime, String.valueOf(id), fromAddr, toAddr, subject, true, true, teHost, smtpHost, smtpPort, phantomJsPath);
```

Comment:
1. What if there is no anomalies generated during the reply? If my understanding correct, there will be no email send. But how user distinguish 'no anomaly detected' vs 'replay is not run'?

2. Should we use slightly different email template for this replay notification email?